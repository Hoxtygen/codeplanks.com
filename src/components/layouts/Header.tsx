import Link from "next/link";
import React, { useEffect, useState } from "react";
import { getCategories } from "../../services";
import { BlogError, PostCategory } from "../../typedefs";

export default function Header() {
  const [categories, setCategories] = useState<PostCategory[]>([]);
  const [error, setError] = useState<BlogError | null>(null);

  useEffect(() => {
    getCategories()
      .then((categories) => setCategories(categories))
      .catch((err: BlogError) => setError(err));
  }, []);
  return (
    <div className="container mx-auto px-10 mb-8">
      <div className="border-b w-full inline-block border-black py-8">
        <div className="md:float-left block">
          <Link href="/">
            <span className="cursor-pointer font-bold text-4xl text-blue">
              TalksofCode
            </span>
          </Link>
        </div>
        <div className="hidden md:float-left md:contents">
          {error && <p>{error.error}</p>}
          {categories.map((category) => (
            <Link key={category.id} href={`/category/${category.slug}`}>
              <span
                data-testid="header-link"
                className="md:float-right mt-2 align-middle text-blue ml-4 font-semibold cursor-pointer"
              >
                {category.name}
              </span>
            </Link>
          ))}
        </div>
      </div>
    </div>
  );
}
