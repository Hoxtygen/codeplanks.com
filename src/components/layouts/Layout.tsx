import React from "react";

import { LayoutProps } from "../../typedefs";
import Footer from "./Footer";
import Header from "./Header";

export default function Layout({ children }: LayoutProps) {
  return (
    <>
      <Header />
      {children}
      <Footer />
    </>
  );
}
