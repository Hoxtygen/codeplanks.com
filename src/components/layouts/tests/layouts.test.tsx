import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Header from "../Header";
import Layout from "../Layout";

describe("Header", () => {
  it("renders Header component", () => {
    const view = render(<Header />);
    const headerText = screen.getByText("TalksofCode");
    expect(view).toMatchSnapshot();
    expect(headerText).toBeInTheDocument();
    expect(headerText).toMatchSnapshot();
  });

  it("renders layout", () => {
    const sub = <button data-testid="btn">Child component</button>;
    const { container } = render(<Layout>{sub}</Layout>);
    expect(container).toMatchSnapshot();
  });
});
