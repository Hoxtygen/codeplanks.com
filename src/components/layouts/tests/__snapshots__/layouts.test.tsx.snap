// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto px-10 py-10"
      >
        <div
          class="border-b w-full inline-block border-black py-4"
        >
          <div
            class="md:float-left block"
          >
            <span
              class="cursor-pointer font-bold text-4xl text-blue"
            >
              CodePlanks
            </span>
            <div
              class="inline-block"
            >
              <button
                class="bg-gray-100 rounded-md border-none border-2 p-4"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="logo"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </button>
            </div>
          </div>
          <div
            class="hidden md:float-left md:contents"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto px-10 py-10"
    >
      <div
        class="border-b w-full inline-block border-black py-4"
      >
        <div
          class="md:float-left block"
        >
          <span
            class="cursor-pointer font-bold text-4xl text-blue"
          >
            CodePlanks
          </span>
          <div
            class="inline-block"
          >
            <button
              class="bg-gray-100 rounded-md border-none border-2 p-4"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </button>
          </div>
        </div>
        <div
          class="hidden md:float-left md:contents"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header renders Header component 2`] = `
<span
  class="cursor-pointer font-bold text-4xl text-blue"
>
  CodePlanks
</span>
`;

exports[`Header renders layout 1`] = `
<div>
  <div
    class="container mx-auto px-10 py-10"
  >
    <div
      class="border-b w-full inline-block border-black py-4"
    >
      <div
        class="md:float-left block"
      >
        <span
          class="cursor-pointer font-bold text-4xl text-blue"
        >
          CodePlanks
        </span>
        <div
          class="inline-block"
        >
          <button
            class="bg-gray-100 rounded-md border-none border-2 p-4"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="logo"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </button>
        </div>
      </div>
      <div
        class="hidden md:float-left md:contents"
      />
    </div>
  </div>
  <button
    data-testid="btn"
  >
    Child component
  </button>
  <div
    class="footer-wrapper bg-black"
  >
    <div
      class="container mx-auto px-10 py-10 text-white flex justify-center items-center"
    >
      <p
        class="text-lg font-semibold"
      >
        Â© codeplanks 2020-2022. All code
         
        <a
          href="http://opensource.org/licenses/MIT"
        >
          MIT License
        </a>
      </p>
    </div>
  </div>
</div>
`;
